tests/cxx/pragma_associated.cc

```diff
@@ -0,0 +1,25 @@
+//===--- pragma_associated.cc - test input file for iwyu ------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "tests/cxx/pragma_associated-d1.h"  // IWYU pragma: associated
```

Comment:
I'm not entirely sure if we need to have also `pragma_associated-d2.h` to test that pragma works for multiple files.